woocommerce/woocommerce-android
# Replace Feature Request with Send Feedback using the CrowdSignal Survey

Summary
==========
Fixes issue #2647, this PR delivers the following features:
- Change the previous `Feature Request` option on the Settings view to `Send feedback`
- Trigger the Crowdsignal Feedback Survey view when `Send feedback` is clicked
- Feedback fragments navigation, back stack and exit handled through the Settings Navigation Controller

Changelog
==========
| Description | Commit |
| ------------- | ------------- |
| Remove all `Feature Request` strings and add the `Send Feedback` default one | https://github.com/woocommerce/woocommerce-android/commit/15b720582ca5c0ac46390d91977dadb9cb10a41e |
| Declare and configure the Settings Navigation Graph in order to handle the Survey fragments calls and back stack | https://github.com/woocommerce/woocommerce-android/commit/16dac6bca6075726e05101a3f8db67b049f5ba35 |
| The Settings view context presented issues to handle the WebView rendering, to fix that now the WebView inside the FeedbackSurveyFragment height is defined as `wrap_content` with an explicit vertical orientation | https://github.com/woocommerce/woocommerce-android/commit/48416948543f9f0991c5245eabcb66cb84d4a26a |
| Now that the `FeedbackCompletedFragment` can be called from other Activites than the `MainActivity`, the `Contact us here` button click event handles the `HelpActivity` startup by itself | https://github.com/woocommerce/woocommerce-android/commit/1db677fadcb0dbef92a2d9687ce46a0d884ab26f |

Screenshots
==========
| Click on `Send feedback` option | Wait for the survey loading | Interact with the survey until it's finished | Verify if the expected view is presented and is fully functional | 
| ------------- | ------------- | ------------- | ------------- |
| ![Screenshot_1597972334](https://user-images.githubusercontent.com/5920403/90841385-eea6cc00-e332-11ea-8f79-f8264691b6a2.png) | ![Screenshot_1597812682](https://user-images.githubusercontent.com/5920403/90594829-c5b2f980-e1c1-11ea-9fb4-c8e04118efe3.png) | ![Screenshot_1597891397](https://user-images.githubusercontent.com/5920403/90712144-bcd12f00-e278-11ea-9b6c-b1b014139d00.png) | ![Screenshot_1597812776](https://user-images.githubusercontent.com/5920403/90594835-c9df1700-e1c1-11ea-9b16-63bdb680ed8c.png) |


How to Test
------------

1. Access the `My Store` section from the Woo app
2. On the top right overflow options menu, click in the `Settings` option
3. Verify the availability of a `Send feedback` option instead of the previous `Feature Request` one and click on it
4. Verify that the survey view is called, the loading dialog is visible until the survey is completely loaded
5. Interact with the survey until the conclusion, after finishing it, verify that the custom native Feedback Completed View is called instead of letting the WebView show the default web one.
6. Once inside the Feedback Completed View, verify that each button is working as expected and described


Update release notes:
